java/mapreduce/src/java/org/apache/orc/mapred/OrcMap.java

```diff
@@ -83,4 +83,52 @@ public void readFields(DataInput input) throws IOException {
       put(key, value);
     }
   }
+
+  @Override
+  public int compareTo(OrcMap<K,V> other) {
+    if (other == null) {
+      return -1;
+    }
+    int result = keySchema.compareTo(other.keySchema);
+    if (result != 0) {
+      return result;
+    }
+    result = valueSchema.compareTo(other.valueSchema);
+    if (result != 0) {
+      return result;
+    }
+    Iterator<Map.Entry<K,V>> ourItr = entrySet().iterator();
+    Iterator<Map.Entry<K,V>> theirItr = other.entrySet().iterator();
+    while (ourItr.hasNext() && theirItr.hasNext()) {
+      Map.Entry<K,V> ourItem = ourItr.next();
+      Map.Entry<K,V> theirItem = theirItr.next();
+      K ourKey = ourItem.getKey();
+      K theirKey = theirItem.getKey();
+      int val = ((Comparable<K>) ourKey).compareTo(theirKey);
```

Comment:
Changing the type of K to extend WritableComparable would remove the need for this cast